central/notifiers/syslog/tcp_sender.go

```diff
@@ -82,6 +83,14 @@ func validateRemoteConfig(endpointConfig *storage.Syslog_TCPConfig) (string, err
 		return "", errors.Errorf("invalid port number %d must be between 1 and 65535", port)
 	}
 
+	sysURL := fmt.Sprintf("tcp://%s:%d", endpointConfig.GetHostname(), endpointConfig.GetPort())
+
+	_, err := url.ParseRequestURI(sysURL)
+
+	if err != nil {
+		return "", err
```

Comment:
errors.New with a pretty error message here (assuming all these errors get spewed out ina dialog box on the UI, and going by the way error messages are being handled in the validation cases right above)